[GTK][WPE] Move AcceleratedSurface handling from LayerTreeHost to ThreadedCompositor

carlosgcampos opened the pull request:
#### e17ffe7b5bd3105bdd72617c07655ee4f2d0fdd4
<pre>
[GTK][WPE] Move AcceleratedSurface handling from LayerTreeHost to ThreadedCompositor
<a href="https://bugs.webkit.org/show_bug.cgi?id=281049">https://bugs.webkit.org/show_bug.cgi?id=281049</a>

Reviewed by Miguel Gomez.

The Threaded compositor has a client that is implemented by the
LayerTreeHost and is mainly used to pass messages from the
ThreadedCompositor to the AcceleratedSurface. The ThreadedCompositor
could just create the surface and call it directly instead of going
through the LayerTreeHost.

* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
(WebKit::AcceleratedSurface::AcceleratedSurface):
(WebKit::AcceleratedSurface::frameComplete const):
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.h:
(WebKit::AcceleratedSurface::initialize): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::sizeDidChange):
(WebKit::LayerTreeHost::pauseRendering):
(WebKit::LayerTreeHost::resumeRendering):
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged):
(WebKit::LayerTreeHost::backgroundColorDidChange):
(WebKit::LayerTreeHost::commitSceneState):
(WebKit::LayerTreeHost::preferredBufferFormatsDidChange):
(WebKit::LayerTreeHost::frameComplete): Deleted.
(WebKit::LayerTreeHost::nativeSurfaceHandleForCompositing): Deleted.
(WebKit::LayerTreeHost::didCreateGLContext): Deleted.
(WebKit::LayerTreeHost::willDestroyGLContext): Deleted.
(WebKit::LayerTreeHost::didDestroyGLContext): Deleted.
(WebKit::LayerTreeHost::resize): Deleted.
(WebKit::LayerTreeHost::willRenderFrame): Deleted.
(WebKit::LayerTreeHost::clearIfNeeded): Deleted.
(WebKit::LayerTreeHost::didRenderFrame): Deleted.
(WebKit::LayerTreeHost::didRenderFrameTimerFired): Deleted.
(WebKit::LayerTreeHost::displayDidRefresh): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::ThreadedCompositor::surfaceID const):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::suspend):
(WebKit::ThreadedCompositor::resume):
(WebKit::ThreadedCompositor::setScrollPosition):
(WebKit::ThreadedCompositor::setViewportSize):
(WebKit::ThreadedCompositor::backgroundColorDidChange):
(WebKit::ThreadedCompositor::preferredBufferFormatsDidChange):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::requestComposition):
(WebKit::ThreadedCompositor::frameComplete):
(WebKit::ThreadedCompositor::displayUpdateFired):
(WebKit::ThreadedCompositor::sceneUpdateFinished):
(WebKit::ThreadedCompositor::~ThreadedCompositor): Deleted.
(WebKit::ThreadedCompositor::createGLContext): Deleted.
(WebKit::ThreadedCompositor::updateSceneState): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.h:
(WebKit::ThreadedCompositor::compositingRunLoop const): Deleted.
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::create):
(WebKit::AcceleratedSurfaceDMABuf::AcceleratedSurfaceDMABuf):
(WebKit::AcceleratedSurfaceDMABuf::frameDone):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:
* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.cpp:
(WebKit::AcceleratedSurfaceLibWPE::create):
(WebKit::AcceleratedSurfaceLibWPE::AcceleratedSurfaceLibWPE):
(WebKit::AcceleratedSurfaceLibWPE::finalize):
(WebKit::AcceleratedSurfaceLibWPE::window const):
(WebKit::AcceleratedSurfaceLibWPE::initialize): Deleted.
* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.h:

Canonical link: <a href="https://commits.webkit.org/285007@main">https://commits.webkit.org/285007@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cdc4c29a8b0ee9161a0d19fd8538df7853dc5783

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71046 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50458 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23819 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75150 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22251 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58257 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22069 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56174 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14648 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74112 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45844 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61244 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36616 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42493 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/20592 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/64426 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19047 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/76870 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15278 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18228 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63920 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15322 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61323 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63870 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11982 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5624 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10917 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46259 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1035 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47331 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48614 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47073 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

magomez commented on Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.h:
```diff
@@ -41,28 +41,20 @@ class AcceleratedSurfaceLibWPE final : public AcceleratedSurface {
-    AcceleratedSurfaceLibWPE(WebPage&, Client&);
+    AcceleratedSurfaceLibWPE(WebPage&, Function<void()>&& frameCompleteHandler);
+
+    void initialize();
```
shouldn't this override?

carlosgcampos replied:
No, I removed the virtual initialize from the parent because it's only needed by this class. My initial idea was to remove initialize and move the code to window(), because the caller always does initialize() + window(), but window() is const and we don't want to change that. So, the solution was to keep the initialize() as private non const member that is called from window() with the dirty const cast.

carlosgcampos committed:
[GTK][WPE] Move AcceleratedSurface handling from LayerTreeHost to ThreadedCompositor
https://bugs.webkit.org/show_bug.cgi?id=281049

Reviewed by Miguel Gomez.

The Threaded compositor has a client that is implemented by the
LayerTreeHost and is mainly used to pass messages from the
ThreadedCompositor to the AcceleratedSurface. The ThreadedCompositor
could just create the surface and call it directly instead of going
through the LayerTreeHost.

* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
(WebKit::AcceleratedSurface::AcceleratedSurface):
(WebKit::AcceleratedSurface::frameComplete const):
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.h:
(WebKit::AcceleratedSurface::initialize): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::sizeDidChange):
(WebKit::LayerTreeHost::pauseRendering):
(WebKit::LayerTreeHost::resumeRendering):
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged):
(WebKit::LayerTreeHost::backgroundColorDidChange):
(WebKit::LayerTreeHost::commitSceneState):
(WebKit::LayerTreeHost::preferredBufferFormatsDidChange):
(WebKit::LayerTreeHost::frameComplete): Deleted.
(WebKit::LayerTreeHost::nativeSurfaceHandleForCompositing): Deleted.
(WebKit::LayerTreeHost::didCreateGLContext): Deleted.
(WebKit::LayerTreeHost::willDestroyGLContext): Deleted.
(WebKit::LayerTreeHost::didDestroyGLContext): Deleted.
(WebKit::LayerTreeHost::resize): Deleted.
(WebKit::LayerTreeHost::willRenderFrame): Deleted.
(WebKit::LayerTreeHost::clearIfNeeded): Deleted.
(WebKit::LayerTreeHost::didRenderFrame): Deleted.
(WebKit::LayerTreeHost::didRenderFrameTimerFired): Deleted.
(WebKit::LayerTreeHost::displayDidRefresh): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::ThreadedCompositor::surfaceID const):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::suspend):
(WebKit::ThreadedCompositor::resume):
(WebKit::ThreadedCompositor::setScrollPosition):
(WebKit::ThreadedCompositor::setViewportSize):
(WebKit::ThreadedCompositor::backgroundColorDidChange):
(WebKit::ThreadedCompositor::preferredBufferFormatsDidChange):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::requestComposition):
(WebKit::ThreadedCompositor::frameComplete):
(WebKit::ThreadedCompositor::displayUpdateFired):
(WebKit::ThreadedCompositor::sceneUpdateFinished):
(WebKit::ThreadedCompositor::~ThreadedCompositor): Deleted.
(WebKit::ThreadedCompositor::createGLContext): Deleted.
(WebKit::ThreadedCompositor::updateSceneState): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.h:
(WebKit::ThreadedCompositor::compositingRunLoop const): Deleted.
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::create):
(WebKit::AcceleratedSurfaceDMABuf::AcceleratedSurfaceDMABuf):
(WebKit::AcceleratedSurfaceDMABuf::frameDone):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:
* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.cpp:
(WebKit::AcceleratedSurfaceLibWPE::create):
(WebKit::AcceleratedSurfaceLibWPE::AcceleratedSurfaceLibWPE):
(WebKit::AcceleratedSurfaceLibWPE::finalize):
(WebKit::AcceleratedSurfaceLibWPE::window const):
(WebKit::AcceleratedSurfaceLibWPE::initialize): Deleted.
* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.h:

Canonical link: https://commits.webkit.org/285007@main
